Library/Homebrew/test/spec_helper.rb

```diff
@@ -166,6 +171,23 @@
     skip "Requires network connection." unless ENV["HOMEBREW_TEST_ONLINE"]
   end
 
+  config.before do |example|
+    next if example.metadata.key?(:needs_network)
+    next if example.metadata.key?(:allow_utils_curl)
+
+    DISALLOWED_UTILS_CURL_METHODS.each do |method|
+      allow(Utils::Curl).to receive(method).and_raise(<<~ERROR)
+        Unexpected call to Utils::Curl.#{method} without setting :needs_network.
```

Comment:
```suggestion
        Unexpected call to Utils::Curl.#{method} without setting :needs_network or : needs_curl.
```

whatever name we decide on